internal/informer/informer.go

```diff
@@ -205,10 +223,9 @@ func NewInformer[T runtime.Object](ctx context.Context, opts ...InformerOption[T
 		},
 	})
 	if err != nil {
-		return nil, err
+		return err
 	}
-	i.ctlCh = make(chan struct{})
-	return i, nil
+	return nil
```

Comment:
Nit: We can return the error directly instead of `if err != nil { return err }`

Reply:
Next time :)